sentry-core/src/session.rs

```diff
@@ -0,0 +1,316 @@
+//! Release Health Sessions
+//!
+//! https://develop.sentry.dev/sdk/sessions/
+
+use std::borrow::Cow;
+use std::sync::Arc;
+use std::time::Instant;
+
+use crate::protocol::{Event, Level, User};
+use crate::scope::StackLayer;
+use crate::types::{DateTime, Utc, Uuid};
+
+/// Represents the status of a session.
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum SessionStatus {
+    Ok,
+    Crashed,
+    #[allow(dead_code)]
+    Abnormal,
+    Exited,
+}
+
+pub enum SessionUpdate {
+    NeedsFlushing(Session),
+    Unchanged,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct Session {
+    session_id: Uuid,
+    status: SessionStatus,
+    errors: usize,
```

Comment:
why tie this to pointer size instead of u64?  more efficient?

Reply:
doesn’t really matter I guess, upstream does not dictate any type, and for those cases `usize` is the most appropriate I would say.